pkg/virt-controller/watch/export/export.go

```diff
@@ -0,0 +1,871 @@
+				return true, nil
+			}
+		}
+		return false, nil
+	}
+}
+
+func (ctrl *VMExportController) getExportSecretName(ownerPod *corev1.Pod) string {
+	var certSecretName string
+	for _, volume := range ownerPod.Spec.Volumes {
+		if volume.Name == certificates {
+			certSecretName = volume.Secret.SecretName
+		}
+	}
+	return certSecretName
+}
+
+func (ctrl *VMExportController) getExportServiceName(vmExport *exportv1.VirtualMachineExport) string {
+	// TODO: Ensure name is not too long
+	return fmt.Sprintf("%s-%s", exportPrefix, vmExport.Name)
+}
+
+func (ctrl *VMExportController) getExportPodName(vmExport *exportv1.VirtualMachineExport) string {
+	// TODO: Ensure name is not too long
+	return fmt.Sprintf("%s-%s", exportPrefix, vmExport.Name)
+}
+
+func (ctrl *VMExportController) getOrCreateExportService(vmExport *exportv1.VirtualMachineExport) (*corev1.Service, error) {
+	if service, err := ctrl.Client.CoreV1().Services(vmExport.Namespace).Get(context.Background(), ctrl.getExportServiceName(vmExport), metav1.GetOptions{}); err != nil && !errors.IsNotFound(err) {
+		return nil, err
+	} else if service != nil && err == nil {
+		return service, nil
+	}
+	service := ctrl.createServiceManifest(vmExport)
+	log.Log.V(3).Infof("Creating new exporter service %s/%s", service.Namespace, service.Name)
+	ctrl.Recorder.Eventf(vmExport, corev1.EventTypeNormal, serviceCreatedEvent, "Created service %s/%s", service.Namespace, service.Name)
+	return ctrl.Client.CoreV1().Services(vmExport.Namespace).Create(context.Background(), service, metav1.CreateOptions{})
+}
+
+func (ctrl *VMExportController) createServiceManifest(vmExport *exportv1.VirtualMachineExport) *corev1.Service {
+	service := &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ctrl.getExportServiceName(vmExport),
+			Namespace: vmExport.Namespace,
+			OwnerReferences: []metav1.OwnerReference{
+				*metav1.NewControllerRef(vmExport, schema.GroupVersionKind{
+					Group:   exportv1.SchemeGroupVersion.Group,
+					Version: exportv1.SchemeGroupVersion.Version,
+					Kind:    "VirtualMachineExport",
+				}),
+			},
+		},
+		Spec: corev1.ServiceSpec{
+			Ports: []corev1.ServicePort{
+				{
+					Protocol: "TCP",
+					Port:     443,
+					TargetPort: intstr.IntOrString{
+						Type:   intstr.Int,
+						IntVal: 8443,
+					},
+				},
+			},
+			Selector: map[string]string{
+				exportServiceLabel: vmExport.Name,
+			},
+		},
+	}
+	return service
+}
+
+func (ctrl *VMExportController) getOrCreateExporterPod(vmExport *exportv1.VirtualMachineExport, pvcs []*corev1.PersistentVolumeClaim) (*corev1.Pod, error) {
+	pod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ctrl.getExportPodName(vmExport),
+			Namespace: vmExport.Namespace,
+		},
+	}
+	log.Log.V(3).Infof("Checking if pod exist: %s/%s", pod.Namespace, pod.Name)
+	if pod, err := ctrl.Client.CoreV1().Pods(vmExport.Namespace).Get(context.Background(), pod.Name, metav1.GetOptions{}); err != nil && !errors.IsNotFound(err) {
```

Comment:
Should not be using the API here, rather an informer

Reply:
Done